demos/projects/NXP/mimxrt1060/port/azure_iot_flash_platform.c

```diff
@@ -37,81 +40,161 @@ static AzureIoTResult_t prvBase64Decode( uint8_t * base64Encoded,
+    else
+    {
+        AZLogError( ( "Invalid image position! Will write to image 2" ) );
+        pxAduImage->xUpdatePartition = FLASH_AREA_IMAGE_2_OFFSET;
+    }
+
+    ulPrimask = DisableGlobalIRQ();
+    xStatus = sfw_flash_erase( pxAduImage->xUpdatePartition, FLASH_AREA_IMAGE_1_SIZE );
+    EnableGlobalIRQ( ulPrimask );
+
+    if( xStatus )
+    {
+        AZLogError( ( "Error erasing flash.\r\n" ) );
+        xResult = eAzureIoTErrorFailed;
+    }
+
+    return xResult;
 }
 
 int64_t AzureIoTPlatform_GetSingleFlashBootBankSize()
 {
-    /* TODO: Fill in returning of flash bank size (to make sure we don't install an image larger than that) */
-
-    LogInfo( ( "AzureIoTPlatform_GetSingleFlashBootBankSize()\r\n" ) );
-
-    return INT64_MAX;
+    return FLASH_AREA_IMAGE_1_SIZE;
 }
 
 AzureIoTResult_t AzureIoTPlatform_WriteBlock( AzureADUImage_t * const pxAduImage,
                                               uint32_t ulOffset,
                                               uint8_t * const pData,
                                               uint32_t ulBlockSize )
 {
-    ( void ) pxAduImage;
-    ( void ) ulOffset;
-    ( void ) pData;
-    ( void ) ulBlockSize;
+    uint32_t pucNextWriteAddr = pxAduImage->xUpdatePartition + ulOffset;
+    AzureIoTResult_t xResult = eAzureIoTSuccess;
+    uint32_t ulPrimask;
+    status_t xStatus;
 
-    /* TODO: Fill in to write pData of size ulBlockSize to ulOffset in memory. */
+    ulPrimask = DisableGlobalIRQ();
+    xStatus = sfw_flash_write( pucNextWriteAddr, pData, ulBlockSize );
+    EnableGlobalIRQ( ulPrimask );
 
-    LogInfo( ( "AzureIoTPlatform_WriteBlock(): offset %i\r\n", ulOffset ) );
+    if( xStatus )
+    {
+        AZLogError( ( "Error writing to flash.\r\n" ) );
+        xResult = eAzureIoTErrorFailed;
+    }
 
-    return eAzureIoTSuccess;
+    return xResult;
 }
 
 AzureIoTResult_t AzureIoTPlatform_VerifyImage( AzureADUImage_t * const pxAduImage,
                                                uint8_t * pucSHA256Hash,
                                                uint32_t ulSHA256HashLength )
 {
-    ( void ) pxAduImage;
-    ( void ) pucSHA256Hash;
-    ( void ) ulSHA256HashLength;
+    int xResult;
+    uint32_t ulOutputSize;
+    uint32_t ulReadSize;
+    uint32_t ulPrimask;
 
-    /* TODO: Fill in to verify bytes written to flash bank match the SHA256 given by pucSHA256Hash. */
+    AZLogInfo( ( "Base64 Encoded Hash from ADU: %.*s", ulSHA256HashLength, pucSHA256Hash ) );
+    xResult = prvBase64Decode( pucSHA256Hash, ulSHA256HashLength, ucDecodedManifestHash, azureiotflashSHA_256_SIZE, ( size_t * ) &ulOutputSize );
 
-    LogInfo( ( "AzureIoTPlatform_VerifyImage()\r\n" ) );
+    if( xResult != eAzureIoTSuccess )
+    {
+        AZLogError( ( "Unable to decode image hash SHA256\r\n" ) );
+        return eAzureIoTErrorFailed;
+    }
 
-    return eAzureIoTSuccess;
+    mbedtls_md_context_t ctx;
+    mbedtls_md_type_t md_type = MBEDTLS_MD_SHA256;
+
+    mbedtls_md_init( &ctx );
```

Comment:
Nit: it would be nice if in the future we could extract the mbedTLS crypto into a sample common file if it's the same for other boards.

Reply:
Agree! The asserts work creates a common file for the samples that would facilitate this well.